hil/ext/switches/juniper.py

```diff
@@ -0,0 +1,294 @@
+        `o_session`: session object with connection to switch already open.
+        `set_command`: Multi-line `set` method based command string.
+        """
+        jun = o_session
+        jun.cfg.load(set_command, format="set")
+
+    def _jinja_load_config(self, o_session, j_template, var_dict):
+        """ Load configuration changes using the 'jinja_template' method.
+        `o_session`: session object with connection to switch already open.
+       `j_template`: jinja2 compliant configuration template. Samples is 
+       available at `junos/jinja_templates/`.
+       `var_dict`: Dictionary of variables to be used in the jinja template
+       passed with `j_template`, that will load the specific configuration 
+       change in the switch.
+       """
+        jun = o_session
+        jun.cfg.load(
+               template_path=j_template, template_vars=var_dict,
+               format='text'
+               )
+
+
+    def _set_commit_config(self, set_command):
+        """ Changes the switch config by using only 'set' method.
+        `set_command`: 'set' method based command string.
+        Note: Some configurations changes require both 'set'
+        and 'jinja_template' method. In those cases use combination of functions
+        `_set_load_config`, `_jinja_load_config` as needed finally commit using
+        `_commit_config`. 
+        """
+
+        jun = self.session()
+        try:
+            with jun:
+                self._set_load_config(jun, set_command)
+                self._commit_config(jun)
+        except ConnectError as err:
+                print("connat connect to device: {0}".format(err))
+                sys.exit(1)
+
+    def _interface_info(self, port):
+        """Fetches latest committed configuration information about `port`"""
+        jun = self.session()
+        try: 
+            with jun:
+                interface_config = InterfaceConfigTables(jun)
+                interface_config.get(interface= port, options = {'database':'committed'})
+                ic_dict = json.loads(interface_config.to_json())
+        except ConnectError as err:
+                print("connat connect to device: {0}".format(err))
+                sys.exit(1)
+
+        return ic_dict
+
+    def _trunk_mode(self, port):
+        """Returns True if the port is in trunk mode else returns False. """
+        port_info = self._interface_info(port)
+        return port_info[port]['trunk_mode']
+
+    def _set_mode(self, port, mode):
+        """ Will set a given port to 
+        trunk mode or access mode based on the `mode` flag.
+        """
+        if mode == "trunk":
+            set_mode = """
+            set interfaces {port} unit 0 family ethernet-switching interface-mode trunk
+        """.format(port=port)
+            self._set_commit_config(set_mode)
+        else: 
+            self._remove_all_vlans_port(port)
+
+        #x = "set interfaces et-0/0/08:0 unit 0 family ethernet-switching interface-mode trunk"
```

Comment:
commented out code should be removed

Reply:
It is not commented out code. It is left as an example for future maintainers.